test/test_integration_management.cxx

```diff
@@ -1004,3 +1005,252 @@ TEST_CASE("integration: query index management", "[integration]")
+            index.params_json = couchbase::utils::json::generate(
+              tao::json::value{ { "targets", { { index1_name, tao::json::empty_object }, { index2_name, tao::json::empty_object } } } });
+            couchbase::operations::management::search_index_upsert_request req{};
+            req.index = index;
+            auto resp = test::utils::execute(integration.cluster, req);
+            REQUIRE_FALSE(resp.ctx.ec);
+        }
+
+        {
+            couchbase::operations::management::search_index_get_request req{};
+            req.index_name = index1_name;
+            auto resp = test::utils::execute(integration.cluster, req);
+            REQUIRE_FALSE(resp.ctx.ec);
+            REQUIRE(resp.index.name == index1_name);
+            REQUIRE(resp.index.type == "fulltext-index");
+        }
+
+        {
+            couchbase::operations::management::search_index_get_request req{};
+            req.index_name = "missing_index";
+            auto resp = test::utils::execute(integration.cluster, req);
+            REQUIRE(resp.ctx.ec == couchbase::error::common_errc::index_not_found);
+        }
+
+        {
+            couchbase::operations::management::search_index_get_all_request req{};
+            auto resp = test::utils::execute(integration.cluster, req);
+            REQUIRE_FALSE(resp.ctx.ec);
+            REQUIRE_FALSE(resp.indexes.empty());
+        }
+
+        {
+            couchbase::operations::management::search_index_drop_request req{};
+            req.index_name = index1_name;
+            auto resp = test::utils::execute(integration.cluster, req);
+            REQUIRE_FALSE(resp.ctx.ec);
+        }
+
+        {
+            couchbase::operations::management::search_index_drop_request req{};
+            req.index_name = index2_name;
+            auto resp = test::utils::execute(integration.cluster, req);
+            REQUIRE_FALSE(resp.ctx.ec);
+        }
+
+        {
+            couchbase::operations::management::search_index_drop_request req{};
+            req.index_name = alias_name;
+            auto resp = test::utils::execute(integration.cluster, req);
+            REQUIRE_FALSE(resp.ctx.ec);
+        }
+
+        couchbase::operations::management::search_index_drop_request req{};
+        req.index_name = "missing_index";
+        auto resp = test::utils::execute(integration.cluster, req);
+        REQUIRE(resp.ctx.ec == couchbase::error::common_errc::index_not_found);
+    }
+
+    SECTION("upsert index no name")
+    {
+        couchbase::operations::management::search_index index;
+        index.type = "fulltext-index";
+        index.source_type = "couchbase";
+        index.source_name = integration.ctx.bucket;
+        couchbase::operations::management::search_index_upsert_request req{};
+        req.index = index;
+        auto resp = test::utils::execute(integration.cluster, req);
+        REQUIRE(resp.ctx.ec == couchbase::error::common_errc::invalid_argument);
+    }
```

Comment:
gocb has this test. Do we want to return early with invalid_argument if the name is empty?

Reply:
yes, I think it makes sense. Just like with analytics links we can validate it and return error. I can add the fix for it